test/unit/php/includes/core/classes/class-test-event.php

```diff
@@ -429,7 +425,56 @@ public function test_get_calendar_links(): void {
 				'link' => $expected_yahoo_link,
 			),
 		);
+		*/
+		$output  = $event->get_calendar_links();
+		$expects = array(
+			'google'  => array(
+				'name' => 'Google Calendar',
+				'link' => home_url( '/?gatherpress_event=unit-test-event&gatherpress_calendars=google-calendar' ),
+			),
+			'ical'    => array(
+				'name'     => 'iCal',
+				'download' => home_url( '/?gatherpress_event=unit-test-event&gatherpress_calendars=ical' ),
+			),
+			'outlook' => array(
+				'name'     => 'Outlook',
+				'download' => home_url( '/?gatherpress_event=unit-test-event&gatherpress_calendars=outlook' ),
+			),
+			'yahoo'   => array(
+				'name' => 'Yahoo Calendar',
+				'link' => home_url( '/?gatherpress_event=unit-test-event&gatherpress_calendars=yahoo-calendar' ),
+			),
+		);
+
+		$this->assertSame( $expects, $output );
 
+		// Update permalink structure to '/%postname%/'.
+		update_option( 'permalink_structure', '/%postname%/' );
+
+		// Reload the global rewrite rules object to ensure it reflects the changes.
+		global $wp_rewrite;
+		$wp_rewrite->init();
+		flush_rewrite_rules();
+
+		$output  = $event->get_calendar_links();
+		$expects = array(
+			'google'  => array(
+				'name' => 'Google Calendar',
+				'link' => home_url( '/event/unit-test-event/google-calendar' ),
+			),
+			'ical'    => array(
+				'name'     => 'iCal',
+				'download' => home_url( '/event/unit-test-event/ical' ),
+			),
+			'outlook' => array(
+				'name'     => 'Outlook',
+				'download' => home_url( '/event/unit-test-event/outlook' ),
+			),
+			'yahoo'   => array(
+				'name' => 'Yahoo Calendar',
+				'link' => home_url( '/event/unit-test-event/yahoo-calendar' ),
+			),
+		);
 		$this->assertSame( $expects, $output );
```

Comment:
I tried to change the permalink settings here, but failed.
I wanted to test the rewritten and the _ugly_ permalinks in the same test.

Why did my attempts not work (within the tests)?